RELEASE.md

```diff
@@ -0,0 +1,43 @@
+# Releasing connect-go
+
+This document outlines how to create a release of connect-go.
+
+1. Clone the repo, ensuring you have the latest main.
+
+2. On a new branch, open [connect.go](connect.go) and change the `Version` constant. Do not just remove the `-dev` suffix: look at the release history and the unreleased commits to choose a new semantic version number.
+
+   ```patch
+   -const Version = "1.14.0-dev"
+   +const Version = "1.14.0"
+   ```
+
+3. Check for changes in [cmd/protoc-gen-connect-go/main.go](cmd/protoc-gen-connect-go/main.go) to use an appropriate version restriction that matches the current release. A constant `IsAtLeastVersionX_Y_Z` should be defined in [connect.go](connect.go) if generated code has begun using a new API. Update as required ([Example PR #496](https://github.com/connectrpc/connect-go/pull/496)).
```

Comment:
```suggestion
3. Check [cmd/protoc-gen-connect-go/main.go](cmd/protoc-gen-connect-go/main.go) to ensure that if a new `IsAtLeastVersionX_Y_Z` constant was added since the last release, it is the same version number that you are currently releasing. If a new constant was added since the last release but it doesn't have the same version number as what you are about to release, then update the name to match the version you are about to release. [Example PR #496](https://github.com/connectrpc/connect-go/pull/496).
```

Reply:
I think this might deviate from the intent a bit - I might read these instructions and add a new constant all the time, as part of doing a release. In general, adding new constants is an exception not a rule. I don't have a good suggestion here though.

Reply:
What was the intent of the existing Notion doc if not what I suggested here?
I agree with Ed from earlier comment thread that a constant like this should get introduced in the original PR, not the release PR.

In any case, if we agree that constants like this are rare, and the error we are checking here for is rarer, then I think it is ok to delete this step from the release process.

Reply:
I think the `if a new IsAtLeastVersionX_Y_Z` is not quite right as they might have forgotten to add the version restriction. We need to check `main.go` for any use of connect-go that would fail in a surprising way if the version restriction wasn't added. The wording from protobuf-go: https://github.com/protocolbuffers/protobuf-go/blob/9f73929456f40461a01f2375d7dbf875126bd6ce/internal/version/version.go#L29-L31

What about:

> Check for any new changes to the generator in [cmd/protoc-gen-connect-go/main.go](cmd/protoc-gen-connect-go/main.go) that rely on new functionality in the library. If present, ensure a new constant `IsAtLeastVersionX_Y_Z` has been defined in [connect.go](connect.go) that matches the current release. Update as needed ([Example PR #496](https://github.com/connectrpc/connect-go/pull/496)).

Reply:
> We need to check main.go for any use of connect-go that would fail in a surprising way if the version restriction wasn't added.

I think this reduces to the person doing the release needs to code review all changes that have landed, which, imo, is not something we should be asking the person doing the release to do. The best time to catch whether a new constant needs to be added is in the PR that introduces the change. The only thing thing the release person should do, if anything, is ensure the actual release number matches the expected release number at the time the constant was added (which is the docs that I am suggestion).

If we DO want the person doing the release to effectively code review all changes to look for whether `IsAtLeastVersionX_Y_Z` has been correctly added or not, then I like the wording in Ed's most recent comment:

> Check for any new changes to the generator in [cmd/protoc-gen-connect-go/main.go](https://github.com/connectrpc/connect-go/pull/cmd/protoc-gen-connect-go/main.go) that rely on new functionality in the library. If present, ensure a new constant IsAtLeastVersionX_Y_Z has been defined in [connect.go](https://github.com/connectrpc/connect-go/pull/connect.go) that matches the current release. Update as needed (https://github.com/connectrpc/connect-go/pull/496).

Reply:
I updated my original suggestion to try to make it more clear this is a rare case. I think we should apply this suggestion and have @jhump review this.